src/routes/transactions/helpers/kiln-native-staking.helper.ts

```diff
@@ -24,153 +16,51 @@ export class KilnNativeStakingHelper {
   private static readonly WITHDRAW_SIGNATURE =
     'function batchWithdrawCLFee(bytes) external';
 
-  constructor(
-    private readonly transactionFinder: TransactionFinder,
-    @Inject(IStakingRepository)
-    private readonly stakingRepository: IStakingRepository,
-  ) {}
-
-  public async findDepositTransaction(args: {
-    chainId: string;
-    to?: `0x${string}`;
-    data: `0x${string}`;
-  }): Promise<{ to: `0x${string}`; data: `0x${string}` } | null> {
-    return this.findNativeStakingTransaction({
-      signature: KilnNativeStakingHelper.DEPOSIT_SIGNATURE,
-      ...args,
-    });
-  }
+  constructor(private readonly transactionFinder: TransactionFinder) {}
 
-  public async findValidatorsExitTransaction(args: {
+  public findDepositTransaction(args: {
     chainId: string;
     to?: `0x${string}`;
     data: `0x${string}`;
-  }): Promise<{ to: `0x${string}`; data: `0x${string}` } | null> {
-    return this.findNativeStakingTransaction({
-      signature: KilnNativeStakingHelper.VALIDATORS_EXIT_SIGNATURE,
-      ...args,
-    });
+    value: string;
+  }): { to?: `0x${string}`; data: `0x${string}`; value: string } | null {
+    const selector = toFunctionSelector(
+      KilnNativeStakingHelper.DEPOSIT_SIGNATURE,
+    );
+    return this.transactionFinder.findTransaction(
+      (transaction) => transaction.data.startsWith(selector),
+      args,
+    );
   }
 
-  public async findWithdrawTransaction(args: {
+  public findValidatorsExitTransaction(args: {
     chainId: string;
     to?: `0x${string}`;
     data: `0x${string}`;
-  }): Promise<{ to: `0x${string}`; data: `0x${string}` } | null> {
-    return this.findNativeStakingTransaction({
-      signature: KilnNativeStakingHelper.WITHDRAW_SIGNATURE,
-      ...args,
-    });
+    value: string;
+  }): { to?: `0x${string}`; data: `0x${string}`; value: string } | null {
+    const selector = toFunctionSelector(
+      KilnNativeStakingHelper.VALIDATORS_EXIT_SIGNATURE,
+    );
+    return this.transactionFinder.findTransaction(
+      (transaction) => transaction.data.startsWith(selector),
+      args,
+    );
   }
 
-  private async findNativeStakingTransaction(args: {
-    signature: string;
+  public findWithdrawTransaction(args: {
     chainId: string;
     to?: `0x${string}`;
     data: `0x${string}`;
-  }): Promise<{ to: `0x${string}`; data: `0x${string}` } | null> {
-    const transaction = this.transactionFinder.findTransaction(
-      (transaction) =>
-        transaction.data.startsWith(toFunctionSelector(args.signature)),
+    value: string;
+  }): { to?: `0x${string}`; data: `0x${string}`; value: string } | null {
+    const selector = toFunctionSelector(
+      KilnNativeStakingHelper.WITHDRAW_SIGNATURE,
+    );
+    return this.transactionFinder.findTransaction(
+      (transaction) => transaction.data.startsWith(selector),
       args,
     );
```

Comment:
When writing the tests, I realised we also check the deployment when mapping and fallback to a "standard" transaction there. This is therefore unnecessary, and now covered by tests.